test/helpers/form-monkey.ts

```diff
@@ -0,0 +1,261 @@
+import { GenericTestContext } from 'ava';
+import { IInputMetadata } from '../../app/components/shared/inputs';
+import { SpectronClient } from 'spectron';
+import { sleep } from './sleep';
+import { cloneDeep, isMatch } from 'lodash';
+
+interface IFormMonkeyFillOptions {
+  metadata?: Dictionary<IInputMetadata>
+}
+
+interface IUIInput {
+  id: string;
+  type: string;
+  name: string;
+  selector: string;
+}
+
+/**
+ * helper for simulating user input into SLOBS forms
+ */
+export class FormMonkey {
+
+  private client: SpectronClient;
+
+  constructor(private t: GenericTestContext<any>, private showLogs = false) {
+    this.client = t.context.app.client;
+  }
+
+  async getInputs(formName: string): Promise<IUIInput[]> {
+    const formSelector = `form[name=${formName}]`;
+
+    if (!await this.client.isExisting(formSelector)) {
+      throw new Error(`form not found: ${formName}`);
+    }
+
+    const result = [];
+    const $inputs = await this.client.$$(`${formSelector} [data-role=input]`);
```

Comment:
is there a reason for prefixing `inputs` with `$` here? i get that it's necessary to interact with the api using the stringified format but it feels a little awkward when you do something like `${$inputs.length}`

Reply:
$ at the beginning of the variable is just a codestyle that means that it's a reference to the input element in DOM and not an input value. Not a big deal here btw.